test/e2e/storage/testsuites/provisioning.go

```diff
@@ -0,0 +1,370 @@
+	DelayBinding       bool
+	ClaimSize          string
+	ExpectedSize       string
+	PvCheck            func(volume *v1.PersistentVolume) error
+	NodeName           string
+	SkipWriteReadCheck bool
+	VolumeMode         *v1.PersistentVolumeMode
+}
+
+type provisioningTestSuite struct {
+	tsInfo TestSuiteInfo
+}
+
+var _ TestSuite = &provisioningTestSuite{}
+
+// InitProvisioningTestSuite returns provisioningTestSuite that implements TestSuite interface
+func InitProvisioningTestSuite() TestSuite {
+	return &provisioningTestSuite{
+		tsInfo: TestSuiteInfo{
+			name: "provisioning",
+			testPatterns: []testpatterns.TestPattern{
+				testpatterns.DefaultFsDynamicPV,
+			},
+		},
+	}
+}
+
+func (p *provisioningTestSuite) getTestSuiteInfo() TestSuiteInfo {
+	return p.tsInfo
+}
+
+func (p *provisioningTestSuite) skipUnsupportedTest(pattern testpatterns.TestPattern, driver drivers.TestDriver) {
+}
+
+func createProvisioningTestInput(pattern testpatterns.TestPattern, resource provisioningTestResource) provisioningTestInput {
+	driver := resource.driver
+	dInfo := driver.GetDriverInfo()
+	f := dInfo.Framework
+
+	input := provisioningTestInput{
+		testCase: StorageClassTest{
+			ClaimSize:    resource.claimSize,
+			ExpectedSize: resource.claimSize,
+		},
+		cs:    f.ClientSet,
+		pvc:   resource.pvc,
+		sc:    resource.sc,
+		dInfo: dInfo,
+	}
+
+	if dInfo.Config.ClientNodeName != "" {
+		input.testCase.NodeName = dInfo.Config.ClientNodeName
+	}
+
+	return input
+}
+
+func (p *provisioningTestSuite) execTest(driver drivers.TestDriver, pattern testpatterns.TestPattern) {
+	Context(getTestNameStr(p, pattern), func() {
+		var (
+			resource     provisioningTestResource
+			input        provisioningTestInput
+			needsCleanup bool
+		)
+
+		BeforeEach(func() {
+			needsCleanup = false
+			// Skip unsupported tests to avoid unnecessary resource initialization
+			skipUnsupportedTest(p, driver, pattern)
+			needsCleanup = true
+
+			// Setup test resource for driver and testpattern
+			resource = provisioningTestResource{}
+			resource.setupResource(driver, pattern)
+
+			// Create test input
+			input = createProvisioningTestInput(pattern, resource)
+		})
+
+		AfterEach(func() {
+			if needsCleanup {
+				resource.cleanupResource(driver, pattern)
+			}
+		})
+
+		testProvisioning(&input)
+	})
+}
+
+type provisioningTestResource struct {
+	driver drivers.TestDriver
+
+	claimSize string
+	sc        *storage.StorageClass
+	pvc       *v1.PersistentVolumeClaim
+}
+
+var _ TestResource = &provisioningTestResource{}
+
+func (p *provisioningTestResource) setupResource(driver drivers.TestDriver, pattern testpatterns.TestPattern) {
+	p.driver = driver
+
+	dInfo := p.driver.GetDriverInfo()
+	f := dInfo.Framework
+	ns := f.Namespace
+	volType := pattern.VolType
+
+	// Setup provisioningTest resource
+	switch volType {
+	case testpatterns.DynamicPV:
+		if dDriver, ok := driver.(drivers.DynamicPVTestDriver); ok {
+			p.sc = dDriver.GetDynamicProvisionStorageClass("")
```

Comment:
do the CSI Drivers have this implemented

Reply:
Not yet. And, it is exactly what we aim to do it in https://github.com/kubernetes/kubernetes/pull/68025 .